### ISSUE TITLE
Key binding causing 2 handlers to be called instead of stopping at the most deeply nested

### ISSUE THREAD

**Issue by rkpatel33** [greena13/react-hotkeys#83]
I have an application where there is a pair of global up/down shortcut handlers, and also a local pair for up/down for when a dropdown is in focus (nested of course in the root component).

I am having a problem where BOTH the local dropdown handler as well as the root handler are being called when they are tied to the same key (up or down). This seems to directly contravene the expected behavior that the most nested component wins and I'm not sure whether I am doing something wrong. From the docs it seems this should not be possible at all, and that at most ONE handler will be called.

For example, if I press `up` when the `AssignToDropdown` is in focus, the handler for both `ASSIGNMENT_DROPDOWN__UP` and `TICKETQUEUE__PREV_TICKET` are called.

If I press `down` both `ASSIGNMENT_DROPDOWN__DOWN` and `TICKETQUEUE__NEXT_TICKET` are called.

The dropdown is created using the `react-boostrap` library. Here are the relevant components:

I have a child dropdown element (again, created using the `react-boostrap`) that is defined as follows in a its render function:

```
export default class AssignToDropdown extends Reflux.Component {
  ....

  localShortcutHandlers = {
    ASSIGNMENT_DROPDOWN__UP: (event) => {
      event.preventDefault();
      // do nothing, block the global up down
      console.log('%c >>>>>>>>>>>>>>>>', 'background: #FFFF00', 'UP!');
    },
    ASSIGNMENT_DROPDOWN__DOWN: (event) => {
      event.preventDefault();
      // do nothing, block the global up down
      console.log('%c >>>>>>>>>>>>>>>>', 'background: #FFFF00', 'DOWN!');
    }
  }

  ... 

  render() {
     ...
      return (
        <HotKeys
          className="al-hotkeys-wrapper"
          handlers={ this.localShortcutHandlers }>
          <DropdownButton
            noCaret
            pullRight
            id={ this.props.id }
            className="al-dropdown-button"
            title={ buttonTitle }
            open={ this.state.open }
            onToggle={ this.handleToggle.bind(this) }
            ref={ el => { this.dropDownButtonRef = el; } }>

            { unassignedMenuItem }
            <MenuItem header>Agents</MenuItem>
            { agentMenuItems }

          </DropdownButton>
        </HotKeys>
      );
    }
}

```

And a root level component wrapped in HotKeys with the following keymap definitions and root level handlers

```
export let globalKeymapMainWindow =  {

  ... 
  // global shortcuts
  'TICKETQUEUE__PREV_TICKET': ['up', 'k'],
  'TICKETQUEUE__NEXT_TICKET': ['down', 'j'],

  // local shortcuts for when the dropdown is in focus
  'ASSIGNMENT_DROPDOWN__UP': 'up',
  'ASSIGNMENT_DROPDOWN__DOWN': 'down'

};


// root level handlers

export let globalShortcutHandlers = {
  ...

  'TICKETQUEUE__PREV_TICKET': (event) => {
    shortcutEvents.selectPrevTicketEventEmit();
  },
  'TICKETQUEUE__NEXT_TICKET': (event) => {
    shortcutEvents.selectNextTicketEventEmit();
  },

  ...
};
```

**Comment by rkpatel33** [comment]
To add to this, it seems it has something to with the react-bootstrap dropdown component, which seems to have its own up/down shortcut handling when it has focus.

If I create a dummy action `DUMMY_SHORTCUT` tied to `cmd+shift+t` and create a global and local handler, the local handler wins and does not bubble to the global, as expected.

**Comment by greena13** [comment]
Hello, rkpatel33.

Thank you for the in-depth write-up of your issue. I agree that what you are describing _sounds_ like unexpected behaviour that contradicts the documented functionality.

It's been quite a while since I worked on this module - I am hoping to return to active development in a month - so I am a little rusty on the particulars of how it works.

What happens if you rename both the global and local action to the same thing? Do you still see both the local and global handlers being called?

**Comment by rkpatel33** [comment]
Yes, even the double trigger happens even if you name them the same thing. I think somehow the event is "escaping" or emitted from from the react-bootstrap dropdown which, based on some investigating of that codebase, seems to be an uncontrolled component.

**Comment by greena13** [comment]
Did you end up getting to the bottom of this, @rkpatel33?

**Comment by rkpatel33** [comment]
Hi, no I have not gotten to the bottom of it yet, I had to turn my attention to other things, but return to this code this week so will have another go at it. But I was really at a dead end when I posted so will likely try to find another path forward. Will let you know here later in the week.

**Comment by greena13** [comment]
This will be fixed in [v2.0.0-pre1](https://github.com/greena13/react-hotkeys/releases/tag/v2.0.0-pre1) when it's released, soon.
